In [1]:
#Load Libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns

In [2]:
import numpy as np
import statsmodels.api as sm
import pandas as pd

In [3]:
#Load Dataset
Diabetes=pd.read_csv('./diabetes.csv')
Diabetes.head()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1

In [4]:
# defining the dependent and independent variables
Xtrain = Diabetes[['Pregnancies', 'Glucose', 'BloodPressure','SkinThickness', 'Insulin', 'BMI','DiabetesPedigreeFunction', 'Age']]
ytrain = Diabetes[['Outcome']]

In [5]:
from statsmodels.formula.api import logit
fit_logit = logit("Outcome ~ Pregnancies + Glucose+ BloodPressure + SkinThickness + Insulin + BMI  + DiabetesPedigreeFunction + Age", Diabetes).fit()
print(fit_logit.summary())

Optimization terminated successfully.
         Current function value: 0.470993
         Iterations 6
                           Logit Regression Results                           
Dep. Variable:                Outcome   No. Observations:                  768
Model:                          Logit   Df Residuals:                      759
Method:                           MLE   Df Model:                            8
Date:                Mon, 30 Jan 2023   Pseudo R-squ.:                  0.2718
Time:                        20:50:44   Log-Likelihood:                -361.72
converged:                       True   LL-Null:                       -496.74
Covariance Type:            nonrobust   LLR p-value:                 9.652e-54
                               coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------
Intercept                   -8.4047      0.717    -11.728      0.000      -9.809

In the output, ‘Iterations‘ refer to the number of times the model iterates over the data, trying to optimize the model. By default, the maximum number of iterations performed is 35, after which the optimization fails.


### Brief Interpretation

#### Coef

The interpretation of logistic models is different in the manner that the coefficients are understood from the logit perspective.

Each one-unit change in **Pregnancies** will increase the log odds of getting diabetes by 0.1232 or by ~12%

Each additional **DiabetesPedigreeFunction** is associated with an average increase of .9452 in the log odds of being diabetic. 

The log odds for diabetes increases by 0.09 for each unit of **BMI**, 0.03 for each unit of **Glucose**.

**BloodPressure** : The odds value is less than 1, means, that as bloodpressure level of a patient is increasing by one unit, the odds of that patient to have diabetes is decreasing by ~1.33%.

The regression coefficient for glucose is 0.035. This indicate that one unit increase in the glucose concentration will increase the odds of being diabetes-positive by exp(0.035) 1.04 times.https://www.solumaths.com/en/calculator/calculate/exp/0.035

#### P>|z| 

The values in the P>|z| column represent the p-values for each coefficient.

For example:

**Age** has a p-value of .111. Since this value is not less than .05, it means there is not a statistically significant relationship between age and whether or not  a person has diabetes.

**Pregnancies** has a p-value of less than .05, it means there is a statistically significant relationship between number of pregnancies and diabetes.

The p-Values are very important because, we can consider a linear model to be statistically significant only when p-Values are less that the pre-determined statistical significance level, which is ideally 0.05.

When we look at the summary output, we see that there are significant and not significant variables. We can say that it is significant for Pregnancies, glucose, BMI variable. Estimating variables that are not sgnificant can also lead to unsuccessful results. Therefore, analysis should be continued with variables that best explain the diabetes variable. However, stepwise regression should be used to make precise comments about variables. 

It can be seen that only 5 out of the 8 predictors are significantly associated to the outcome. These include: pregnant, glucose, pressure, mass and pedigree. From the logistic regression results, it can be noticed that some variables - SkinThickness, insulin and age - are not statistically significant. Keeping them in the model may contribute to overfitting. Therefore, they should be eliminated. This can be done automatically using statistical techniques, including stepwise regression and penalized regression methods.
